app/controllers/v0/message_drafts_controller.rb

```diff
@@ -4,36 +4,40 @@ class MessageDraftsController < SMController
     def create
       draft = MessageDraft.new(draft_params)
       raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
-      response = client.post_create_message_draft(draft_params)
 
-      render json: response,
+      draft_response = client.post_create_message_draft(draft_params)
+      render json: draft_response,
              serializer: MessageSerializer,
              status: :created
     end
 
     def update
-      draft = MessageDraft.new(draft_params)
-      raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
+      draft = client.post_create_message_draft(draft_params.merge(id: params[:id]))
 
-      client.post_create_message_draft(draft_params.merge(id: params[:id]))
+      # draft will fail validation if its a reply draft.
+      raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
```

Comment:
The exception is raised in the controller, but the message draft in the Draft API needs validation of the following (the history checking was moved out of the controller): 
1. Validation as a draft
2. Validation that its not a reply draft by calling history, which is no longer in the controller logic.
If either of these fail, then the invalid message is returned and it fails and the exception is raised (either because its a reply draft, or an draft with missing required attributes). However, if the draft is valid, the call to MHV is made.